scripts/generate_cloudevent_protos.dart

```diff
@@ -26,55 +27,198 @@ void setupLogging() {
   });
 }
 
-Future<void> generateDartProtos() async {
+Future<bool> generateDartProtos({required String protocExecutable}) async {
   // Ensure to change back to the previous directory securely after operation
-  final currentDirectory = Directory.current.path;
-  try {
-    Directory.current = 'cloudevents';
-    await Process.run('sh', ['scripts/compile.sh']);
-  } finally {
-    // Safely returning to the original directory even if the operation fails
-    Directory.current = currentDirectory;
+  await Process.run(
+    'sh',
+    ['scripts/compile.sh'],
+    workingDirectory: 'cloudevents',
+  );
+
+  // These paths are relative to the 'cloudevents' directory,
+  // which is the current working directory for protoc.
+  const outputPath = './lib/src';
+  final outputDir = Directory(outputPath);
+
+// Clean the outputPath directory if it exists to ensure a clean state for generation.
+// which is the current working directory for protoc.
+  if (!await _ensureCleanOutputDir(outputDir)) {
+    return false;
   }
 
   const protosPath = './cloudevents';
-  const outputPath = './lib/src';
-  final protocCommand = [
-    'protoc',
-    '-I$protosPath/proto',
-    '-I$protosPath/third_party/googleapis',
-    '-I$protosPath/tmp/include',
-    '--dart_out=$outputPath',
+  final protocArgsForFile = <String>[
+    '-I${_toProtocArgPath(path_lib.join(protosPath, 'proto'))}',
```

Comment:
Do you think that you could assign the `_toProtocArgPath` computation(s) to a variable and then interpolate the variable for readability?

```dart
var 3pOutputDirectory = _toProtocArgPath(path_lib.join(protosPath, 'proto'));

final protocArgsForFile = <String>[
  'I${3dOutputDirectory}'
];
```